File: reduks/src/main/java/com/beyondeye/reduks/Reducer.java
package com.beyondeye.reduks;

/**
 * keep this interface in java to leverage kotlin SAM conversion (lambda automatical converted to implementation of java interface)
 * see also https://github.com/reactjs/redux/blob/master/docs/Glossary.md#reducer
 */
public interface Reducer<S> {
    S reduce(S state, Object action);
}

File: reduks/src/main/kotlin/com/beyondeye/reduks/Reducer.kt
package com.beyondeye.reduks

/**
 * single method interface, mainly used because kotlin does not support yet type alias for function types
 * see also https://github.com/reactjs/redux/blob/master/docs/Glossary.md#reducer
 */
interface Reducer<S> {
    fun reduce(state: S, action: Any): S
}
